src/neo4j_genai/retrievers/base.py

```diff
@@ -93,7 +144,20 @@ def search(self, *args: Any, **kwargs: Any) -> RetrieverResult:
         )
 
     @abstractmethod
-    def _get_search_results(self, *args: Any, **kwargs: Any) -> RawSearchResult:
+    def get_search_results(self, *args: Any, **kwargs: Any) -> RawSearchResult:
```

Comment:
I think we can add an example to the docs on how to implement this method, but doesn't have to be part of this PR, wdyt?

Reply:
Yes sure, we can probably find an example to show implementation of a new retriever. For instance a retriever with a specific retrieval query as suggested by Oskar.